ts/ui/lazy/LazyHandler.ts

```diff
@@ -0,0 +1,483 @@
+          state = STATE.COMPILED;
+        } else {
+          math.state(STATE.TYPESET - 1);
+        }
+        math.lazyCompile = math.lazyTypeset = false;
+        math.lazyMarker && this.lazyObserver.unobserve(math.lazyMarker as any as Element);
+      }
+      return state;
+    }
+
+    /**
+     * Mark any TeX items (earlier than the ones in the set) to be compiled.
+     *
+     * @param {LazySet} set   The set of items that are newly visible
+     * @return {boolean}      True if there are TeX items to be typeset
+     */
+    protected compileEarlierItems(set: LazySet): boolean {
+      let math = this.earliestTex(set);
+      if (!math) return false;
+      let compile = false;
+      for (const item of this.math) {
+        if (item === math) break;
+        const earlier = item as LazyMathItem<N, T, D>;
+        if (earlier.lazyTex && earlier.lazyCompile) {
+          earlier.lazyCompile = false;
+          earlier.lazyMarker && this.lazyObserver.unobserve(earlier.lazyMarker as any as Element);
+          earlier.state(STATE.COMPILED - 1);
+          compile = true;
+        }
+      }
+      return compile;
+    }
+
+    /**
+     * Find the earliest TeX math item in the set, if any.
+     *
+     * @param {LazySet} set     The set of newly visble math items
+     * @return {LazyMathItem}   The earliest TeX math item in the set, if any
+     */
+    protected earliestTex(set: LazySet): LazyMathItem<N, T, D> {
+      let min: number = null;
+      let minMath = null;
+      for (const id of set.values()) {
+        const math = this.lazyList.get(id);
+        if (!math.lazyTex) continue;
+        if (min === null || parseInt(id) < min) {
+          min = parseInt(id);
+          minMath = math;
+        }
+      }
+      return minMath;
+    }
+
+    /**
+     * If any of the removed items are observed or in the lazy list, remove them.
+     *
+     * @override
+     */
+    public clearMathItemsWithin(containers: ContainerList<N>) {
+      const items = super.clearMathItemsWithin(containers) as LazyMathItem<N, T, D>[];
+      for (const math of items) {
+        const marker = math.lazyMarker;
+        if (marker) {
+          this.lazyObserver.unobserve(marker as any as Element);
+          this.lazyList.delete(this.adaptor.getAttribute(marker, LAZYID));
+        }
+      }
+      return items;
+    }
+
+  };
+
+}
+
+/*==========================================================================*/
+
+/**
+ * Add lazy typesetting support a Handler instance
+ *
+ * @param {Handler} handler   The Handler instance to enhance
+ * @return {Handler}          The handler that was modified (for purposes of chaining extensions)
+ *
+ * @template N  The HTMLElement node class
+ * @template T  The Text node class
+ * @template D  The Document class
+ */
+export function LazyHandler<N, T, D>(handler: HTMLHandler<N, T, D>): HTMLHandler<N, T, D> {
+  //
+  // Only update the document class if we can handle IntersectionObservers and idle callbacks
```

Comment:
Idle callbacks don't have to be available. You are using timeouts above.

Reply:
Right.  Removed the comment (the test already was only for IntersectionObservers).